picard/ui/scripteditor.py

```diff
@@ -383,14 +509,44 @@ def synchronize_selected_example_lines(current_row, source, target):
     def match_after_to_before(self):
         """Sets the selected item in the 'after' list to the corresponding item in the 'before' list.
         """
-        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_before, self.ui.example_filename_after)
+        self.synchronize_selected_example_lines(self.examples_current_row, self.ui.example_filename_before, self.ui.example_filename_after)
 
     def match_before_to_after(self):
         """Sets the selected item in the 'before' list to the corresponding item in the 'after' list.
         """
-        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_after, self.ui.example_filename_before)
+        self.synchronize_selected_example_lines(self.examples_current_row, self.ui.example_filename_after, self.ui.example_filename_before)
+
+    def delete_script(self):
+        """Removes the currently selected script from the script selection combo box and script list.
+        """
+        if self.confirmation_dialog(_('Are you sure that you want to delete the script?')):
+            idx = self.ui.preset_naming_scripts.currentIndex()
+            self.ui.preset_naming_scripts.blockSignals(True)
+            self.ui.preset_naming_scripts.removeItem(idx)
+            self.ui.preset_naming_scripts.setCurrentIndex(0)
+            self.ui.preset_naming_scripts.blockSignals(False)
+            self.update_scripts_list()
+            self.select_script()
 
     def save_script(self):
+        """Saves changes to the current script to the script list and combo box item.
+        """
+        selected = self.ui.preset_naming_scripts.currentIndex()
+        if selected == 0:
```

Comment:
Why is this index handled differently?

Reply:
I think it's because this specific script is handled separately. The logic as for now is that we still have the config option `file_naming_format`, which is used elsewhere in the application. This is always index 0 here. Then we have separate scripts.

In the scope of this PR I think that behavior is still ok, but actually we should change it. We should completely get rid of `file_naming_format` and instead just store which of the scripts currently is the active one. We introduce a helper function to get the current script, and that is used wherever we need the actual active naming script.

Because we can keep other parts of Picard unchanged I think it's ok to keep the behavior with `file_naming_format` for this PR. We have enough to discuss here already. But afterwards we should actually change this. I guess we then could even get rid of the special index 0 "System" entry here.

Reply:
That was to ensure that changes saved to the "Current file naming script" would be saved to the working version of the config settings for processing when `Make it so!` was triggered.  Changes made to the other user scripts would just be saved to the script object and updated in the user's script list.  To make those changes to the naming script in the config settings, the user needs to use the `Select` button.

I'm not completely happy with having both the `Save` and `Select` buttons because it isn't intuitive how they differ, but I'm struggling to find a better solution.